backend/src/index.ts

```diff
@@ -47,9 +47,17 @@ export class ConnectionManager {
         buffer = buffer.slice(0, 20000)
       }
 
+      let decoded_payload = null
+      // spell-checker: disable-next-line
+      if (topic.match(/spBv1\.0\/[^/]+\/(DDATA|NDATA|NCMD|DCMD|NBIRTH|DBIRTH|NDEATH|DDEATH\/[^/]+\/)/u)) {
```

Comment:
1. The first '/' seems to be wrong. Sparkplug topics don't start with '/'.
2. The device topics have an additional topic level for the device id. The regex expression is missing that topic except for the DDEATH topic type.

You can try this expression: `^spBv1\.0\/[^\/]+\/(N(DATA|CMD|BIRTH|DEATH)\/[^\/]+|D(DATA|CMD|BIRTH|DEATH)\/[^\/]+\/[^\/]+)$`. I have tested that expression only manually.

Reply:
Simplified it a bit as `^spBv1\.0\/[^/]+\/[ND](DATA|CMD|DEATH|BIRTH)\/[^/]+(\/[^/]+)?$`

Does it look correct, have a look at https://regex101.com/r/wwzYQf/1

Reply:
> Simplified it a bit as ^spBv1\.0\/[^/]+\/[ND](DATA|CMD|DEATH|BIRTH)\/[^/]+(\/[^/]+)?$
> 
> Does it look correct?, have a look at https://regex101.com/r/wwzYQf/1

Cool link! I think it's ok. The regex expression is also matching node topics with a device id element, like `spBv1.0/Sparkplug Devices/NBIRTH/JavaScript Edge Node/Emulated Device` or device topics without a device id element like `spBv1.0/Sparkplug Devices/DBIRTH/JavaScript Edge Node`. That is not allowed by the Sparkplug specification but I also don't think, this will cause any troubles.